src/lib.rs

```diff
@@ -41,1346 +53,988 @@ use std::{
-        rng: Option<&mut dyn RngCore>,
-    ) -> Result<BatchLCProof<F, Self>, Self::Error>
-    where
-        Self::Randomness: 'a,
-        Self::Commitment: 'a,
-    {
-        let linear_combinations: Vec<_> = linear_combinations.into_iter().collect();
-        let polynomials: Vec<_> = polynomials.into_iter().collect();
-        let poly_query_set =
-            lc_query_set_to_poly_query_set(linear_combinations.iter().copied(), query_set);
-        let poly_evals = evaluate_query_set(polynomials.iter().copied(), &poly_query_set);
-        let proof = Self::batch_open_individual_opening_challenges(
-            ck,
-            polynomials,
-            commitments,
-            &poly_query_set,
+    ) -> Result<Option<Self::VerifierState>, Self::Error> {
+        let (combined_commitment, combined_value, combined_point) =
+            Self::multi_point_multi_poly_verify_combine(
+                labeled_commitments,
+                query_set,
+                evaluations,
+                multi_point_proof,
+                fs_rng,
+            )?;
+
+        Self::succinct_verify(
+            vk,
+            &combined_commitment,
+            combined_point,
+            combined_value,
+            multi_point_proof.get_proof(),
             fs_rng,
-            rands,
-            rng,
-        )?;
-        Ok(BatchLCProof {
-            proof,
-            evals: Some(poly_evals.values().copied().collect()),
-        })
+        )
     }
 
-    /// Default implementation of Multi point multi LC verify, with random oracle passed from 'outside'.
-    /// Evaluates each of the (non-trivial) LC-polynomials at the query point.
-    /// CAUTION: This is a low-level function to be handled carefully, typically
-    /// presuming that commitments and query_set is already bound to the internal
-    /// state of the Fiat-Shamir rng.
-    /// TODO: rename this function
-    fn check_combinations_individual_opening_challenges<'a>(
-        vk: &Self::VerifierKey,
-        linear_combinations: impl IntoIterator<Item = &'a LinearCombination<F>>,
-        commitments: impl IntoIterator<Item = &'a LabeledCommitment<Self::Commitment>>,
-        eqn_query_set: &QuerySet<F>,
-        eqn_evaluations: &Evaluations<F>,
-        proof: &BatchLCProof<F, Self>,
+    /// Build the multi-poly single-point (random) linear combination.
+    // TODO: let us declare this function as internal.
+    fn single_point_multi_poly_open_combine<'a>(
+        labeled_polynomials: impl IntoIterator<Item = &'a LabeledPolynomial<G::ScalarField>>,
+        labeled_commitments: impl IntoIterator<Item = &'a LabeledCommitment<Self::Commitment>>,
         fs_rng: &mut Self::RandomOracle,
-    ) -> Result<bool, Self::Error>
-    where
-        Self::Commitment: 'a,
-    {
-        let BatchLCProof { proof, evals } = proof;
+        labeled_randomnesses: impl IntoIterator<Item = &'a LabeledRandomness<Self::Randomness>>,
+    ) -> Result<
```

Comment:
Why do we provide the commitments, if we do not compute their linear combination? See my comment below.

Reply:
Removed